interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,301 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var mediaHubPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the the media-hub service. Reserved because
+#  this gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="org.mpris.MediaPlayer2.MediaHub",
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="core.ubuntu.media.Service",
+
+dbus (receive, send)
+	bus=session
+	path=/com/ubuntu/media/Service{,/**}
+	interface=org.freedesktop.DBus**
+	peer=(label=unconfined),
+
+dbus (send)
+	bus=session
+	path=/org/freedesktop/Telepathy/AccountManager
+	interface=org.freedesktop.DBus.Properties
+	member="GetAll",
```

Comment:
This seems wholly unrelated. Why is it needed? This doesn't exist in core so only another snap would be able to provide this (unless this is only needed on classic). Either way, this feels like another interface.